packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -485,7 +495,7 @@ export default {
           for (const hook of reactHooks) {
             // Report an error if a hook may be called more then once.
             // `use(...)` can be called in loops.
-            if (cycled && !isUseIdentifier(hook)) {
+            if ( (cycled || isInsideDoWhileLoop(hook)) && !isUseIdentifier(hook)) {
```

Comment:
We're now considering a hook "cycled" also if it's called within a do/while loop.